testsuite/src/test/java/org/jboss/intersmash/tests/wildfly/web/cache/offload/infinispan/Infinispan2ReplicasCustomCertificateService.java

```diff
@@ -0,0 +1,137 @@
+/**
+* Copyright (C) 2025 Red Hat, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*         http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.jboss.intersmash.tests.wildfly.web.cache.offload.infinispan;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.infinispan.v1.Infinispan;
+import org.infinispan.v1.InfinispanBuilder;
+import org.infinispan.v1.infinispanspec.security.EndpointEncryption;
+import org.infinispan.v1.infinispanspec.security.EndpointEncryptionBuilder;
+import org.infinispan.v2alpha1.Cache;
+import org.jboss.intersmash.application.openshift.OpenShiftApplication;
+import org.jboss.intersmash.application.operator.InfinispanOperatorApplication;
+import org.jboss.intersmash.tests.wildfly.web.cache.offload.infinispan.util.InfinispanCommandLineBasedKeystoreGenerator;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Application descriptor that represents an Infinispan/Red Hat Data Grid service deployed by the related Operator, and
+ * having two replicas.
+ */
+public class Infinispan2ReplicasCustomCertificateService implements InfinispanOperatorApplication, OpenShiftApplication {
+	public static final String INFINISPAN_APP_NAME = "infinispan";
+	public static final String INFINISPAN_CUSTOM_CREDENTIALS_USERNAME = "foo";
+	public static final String INFINISPAN_CUSTOM_CREDENTIALS_PASSWORD = "bar";
+	public static final String INFINISPAN_CUSTOM_CREDENTIALS_SECRET_NAME = "infinispan-custom-credentials-secret";
+	private static final Secret INFINISPAN_CUSTOM_CREDENTIALS_SECRET = buildInfinispanCustomCredentialsSecret();
+	public static final String TLS_KEYSTORE_SECRET_NAME = String.format("%s-custom-tls-secret", INFINISPAN_APP_NAME);
+	public static final String TLS_CERTIFICATE_SECRET_NAME = String.format("%s-custom-cert-secret", INFINISPAN_APP_NAME);
+	public static final String TLS_CERTIFICATE_FILE_NAME = String.format("%s.crt", INFINISPAN_APP_NAME);
+	public static final String STOREPASS = "1234PIPPOBAUDO";
+	public static final String KEYALIAS = "server";
+
+	protected Infinispan infinispan;
+	protected final List<Secret> secrets = new ArrayList<>();
+
+	private static String getEncodedIdentitiesSecretContents(InputStream is) {
```

Comment:
This is already available in [Infinispan2ReplicasService.java](https://github.com/Intersmash/intersmash-tests/blob/main/testsuite/src/test/java/org/jboss/intersmash/tests/wildfly/web/cache/offload/infinispan/Infinispan2ReplicasService.java#L53)

Can you please refactor the code and move such method to utilities class so that it can be used by both the Infinispan service application classes?